app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -98,46 +93,12 @@ class SearchFragment : Fragment() {
         getAutoDisposeObservable<AwesomeBarAction>()
             .subscribe {
                 when (it) {
-                    is AwesomeBarAction.ItemSelected -> transitionToBrowser()
+                    is AwesomeBarAction.ItemSelected -> {
+                        requireComponents.core.sessionManager.selectedSession?.let { session ->
+                            (activity as HomeActivity).openToBrowser(session.id)
+                        }
+                    }
                 }
             }
     }
-
-    // Issue: https://github.com/mozilla-mobile/fenix/issues/626
-    // Currently we were kind of forcing all this logic through the Toolbar Feature.
-    // But since we cannot actually load a page without an available GeckoSession
-    // we have to wait until after we navigate to call the use case.
-    // We should move this logic into a place that makes more sense.
-    private fun load(text: String) {
-        val sessionId = SearchFragmentArgs.fromBundle(arguments!!).sessionId
-        val isPrivate = (activity as HomeActivity).browsingModeManager.isPrivate
-
-        val loadUrlUseCase = if (sessionId == null) {
-            if (isPrivate) {
-                requireComponents.useCases.tabsUseCases.addPrivateTab
-            } else {
-                requireComponents.useCases.tabsUseCases.addTab
-            }
-        } else requireComponents.useCases.sessionUseCases.loadUrl
-
-        val searchUseCase: (String) -> Unit = { searchTerms ->
-            if (sessionId == null) {
-                requireComponents.useCases.searchUseCases.newTabSearch
-                    .invoke(searchTerms, Session.Source.USER_ENTERED, true, isPrivate)
-            } else requireComponents.useCases.searchUseCases.defaultSearch.invoke(searchTerms)
-        }
-
-        if (text.isUrl()) {
-            loadUrlUseCase.invoke(text.toNormalizedUrl())
-        } else {
-            searchUseCase.invoke(text)
-        }
-    }
-
-    private fun transitionToBrowser() {
```

Comment:
I'm not sure if we need a separate `transitionToBrowser` function here since HomeActivity already has one and we want to route logic there for private mode checking.